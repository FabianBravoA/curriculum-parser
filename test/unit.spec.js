'use strict';


const Path = require('path');
const Unit = require('../lib/unit');
const Helpers = require('./helpers');


describe('Unit(path)', () => {

  it('should return a function', () => {
    expect(typeof Unit(Path.join(__dirname, 'fixtures'))).toBe('function');
  });

  describe('Unit(path)(unitDir, cb)', () => {

    it('should parse a unit dir containing a part dir with a README.md', (done) => {
      Unit(Helpers.resolveFixturePath('.'))(
        '00-unit-foo',
        (err, json) => {
          expect(err).toBe(null);
          expect(json).toMatchSnapshot();
          done();
        }
      );
    });

  });

});
