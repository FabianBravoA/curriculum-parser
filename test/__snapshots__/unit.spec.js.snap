// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit(path) Unit(path)(unitDir, cb) should parse a unit dir containing a part dir with a README.md 1`] = `
Array [
  Object {
    "files": Array [
      "README.md",
    ],
    "log": Array [
      Array [
        "error",
        "00-unit-foo/00-part-bar => Could not parse part duration",
      ],
      Array [
        "error",
        "00-unit-foo/00-part-bar => Unkown part format \\"undefined\\". Valid formats: \\"guided\\", \\"self-paced\\"",
      ],
      Array [
        "error",
        "00-unit-foo/00-part-bar => Unkown part type \\"undefined\\". Valid types: \\"read\\", \\"seminar\\", \\"workshop\\", \\"quiz\\", \\"practice\\", \\"other\\"",
      ],
    ],
    "result": Object {
      "body": "<h2 id=\\"acompa-amiento-y-puntos-de-apoyo\\">Acompa&#xF1;amiento y puntos de apoyo</h2>
<p>No te asustes, porque no est&#xE1;s sola, nosotros queremos acompa&#xF1;arte en este proceso y por ello te brindamos tres (3) opciones a trav&#xE9;s de las cuales podremos apoyarte durante el curso:</p>
<h3 id=\\"1-grupo-de-facebook\\">1. Grupo de Facebook</h3>
<p>En Laboratoria creemos mucho en la investigaci&#xF3;n individual y tambi&#xE9;n en el apoyo en grupo. Lo importante es que individualmente entiendas lo que vas haciendo, as&#xED; te apoyes en tus compa&#xF1;eras o amigas. Adem&#xE1;s, recuerda que la mejor manera de aprender es ense&#xF1;ar! No dejes de compartir tus conocimientos con tus compa&#xF1;eras.</p>
<ul>
<li>Lima: <a href=\\"https://facebook.com/groups/LaboratoriaLima20172/?ref=group_cover\\" target=\\"_blank\\">https://facebook.com/groups/LaboratoriaLima20172/?ref=group_cover</a></li>
<li>Ciudad de M&#xE9;xico: <a href=\\"https://facebook.com/groups/LaboratoriaMX20172/?ref=group_cover\\" target=\\"_blank\\">https://facebook.com/groups/LaboratoriaMX20172/?ref=group_cover</a></li>
<li>Santiago de Chile: <a href=\\"https://facebook.com/groups/LaboratoriaCHILE20172/?ref=group_cover\\" target=\\"_blank\\">https://facebook.com/groups/LaboratoriaCHILE20172/?ref=group_cover</a></li>
</ul>
<h3 id=\\"2-horas-de-consulta\\">2. Horas de consulta</h3>
<p>Queremos que sepas que puedes contar con nosotros, pues nuestro principal objetivo es que aprendas. Te invitamos a las horas de consulta presenciales en Laboratoria, donde podr&#xE1;s recibir la ayuda de nuestro staff de profesores. Si no puedes asistir, no te preocupes, pues transmitiremos estas sesiones por Facebook live y luego subiremos los videos a los grupos.</p>
<p>Las horas de consulta por sede son las siguientes:</p>
<ul>
<li>Lima:<ul>
<li>Viernes 14 de julio de 6pm a 8pm</li>
<li>Viernes 21 de julio de 6pm a 8pm</li>
<li>Mi&#xE9;rcoles 26 de julio de 6pm a 8pm</li>
</ul>
</li>
</ul>
<ul>
<li>M&#xE9;xico:<ul>
<li>Viernes 14 de julio de 5pm a 7pm</li>
<li>Mi&#xE9;rcoles 19 de julio de 5pm a 7pm</li>
<li>Mi&#xE9;rcoles 26 de julio de 5pm a 7pm</li>
</ul>
</li>
</ul>
<ul>
<li>Santiago de Chile:<ul>
<li>Jueves 13 de julio de 6pm a 8pm</li>
<li>Jueves 20 de julio de 6pm a 8pm</li>
<li>Jueves 27 de julio de 6pm a 8pm</li>
</ul>
</li>
</ul>
<h3 id=\\"3-punto-principal-de-contacto\\">3. Punto principal de contacto</h3>
<p>Creemos que con el grupo de facebook y las horas de consulta, estar&#xE1;s muy bien acompa&#xF1;ada por nuestros profesores y por todas las postulantes, y que por ambos canales lograr&#xE1;s resolver todas tus dudas.</p>
<p>De todas maneras, te brindamos puntos principales de contacto por sede, para que puedas realizar consultas excepcionales e importantes que no puedas solucionar por los otros medios.</p>
<ul>
<li>Lima:<ul>
<li>Michelle Seguil, <a href=\\"mailto:michelle@laboratoria.la\\" target=\\"_blank\\">michelle@laboratoria.la</a></li>
</ul>
</li>
<li>M&#xE9;xico:<ul>
<li>Jonathan Parra, <a href=\\"mailto:jonathan@laboratoria.la\\" target=\\"_blank\\">jonathan@laboratoria.la</a></li>
<li>Adoraci&#xF3;n Cid, <a href=\\"mailto:adoracion@laboratoria.la\\" target=\\"_blank\\">adoracion@laboratoria.la</a></li>
<li>Grissel Rocha, <a href=\\"mailto:grissel@laboratoria.la\\" target=\\"_blank\\">grissel@laboratoria.la</a></li>
</ul>
</li>
<li>Santiago de Chile:<ul>
<li>Allison Guzm&#xE1;n: <a href=\\"mailto:allison@laboratoria.la\\" target=\\"_blank\\">allison@laboratoria.la</a></li>
<li>Bel&#xE9;n Recabal: <a href=\\"mailto:belen@laboratoria.la\\" target=\\"_blank\\">belen@laboratoria.la</a></li>
<li>Alexandra Neira; <a href=\\"mailto:alexandra@laboratoria.la\\" target=\\"_blank\\">alexandra@laboratoria.la</a></li>
<li>Nadia Tapia: <a href=\\"mailto:nadia@laboratoria.la\\" target=\\"_blank\\">nadia@laboratoria.la</a></li>
<li>Marcia Millanao: <a href=\\"mailto:marcia@laboratoria.la\\" target=\\"_blank\\">marcia@laboratoria.la</a></li>
</ul>
</li>
</ul>
",
      "duration": null,
      "partDir": "00-part-bar",
      "title": "Bienvenida y orientaci√≥n",
      "unitDir": "00-unit-foo",
    },
  },
]
`;
